selfservice/flow/login/hook.go

```diff
@@ -60,10 +60,18 @@ func NewHookExecutor(d executorDependencies) *HookExecutor {
 	return &HookExecutor{d: d}
 }
 
-func (e *HookExecutor) requiresAAL2(r *http.Request, s *session.Session) (*session.ErrAALNotSatisfied, bool) {
+func (e *HookExecutor) requiresAAL2(r *http.Request, s *session.Session, a *Flow) (*session.ErrAALNotSatisfied, bool) {
 	var aalErr *session.ErrAALNotSatisfied
 	err := e.d.SessionManager().DoesSessionSatisfy(r, s, e.d.Config(r.Context()).SessionWhoAmIAAL())
-	return aalErr, errors.As(err, &aalErr)
+	if ok := errors.As(err, &aalErr); !ok {
+		return nil, false
+	}
+
+	if err := aalErr.PassReturnToParameter(a.RequestURL); err != nil {
+		return aalErr, true
```

Comment:
Wasn't this supposed to be:
```suggestion
		return err, true
```
?

Reply:
Thanks! I fixed.